aop/src/main/java/io/micronaut/aop/writer/AopProxyWriter.java

```diff
@@ -527,6 +528,31 @@ public void visitIntroductionMethod(TypedElement declaringBean,
     public void visitAroundMethod(TypedElement beanType,
                                   MethodElement methodElement) {
 
+        final List<MethodElement> overridden = methodElement.getOwningType()
+                .getEnclosedElements(ElementQuery.ALL_METHODS
```

Comment:
You can use `getEnclosedElement` instead of `getEnclosedElements` which returns an `Optional` that would simplify the code